src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -340,33 +369,47 @@ public Response getMeta(String json)
 		catch (JsonSyntaxException e)
 		{
 			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
+			Tags.ERROR.set(span, Boolean.TRUE);
```

Comment:
It could be possible to NPE here if somehow activateScope itself throws as we're not setting local Span until after that in the try-block.  I would recommend refactoring activateScope to something that just activates the span/context, and create the scope per the examples.

We don't control the ScopeManager implementations - Lightstep does - so we shouldn't assume safety any more than we have to.